### PR TITLE
Add rust to the mirroring list

### PR DESCRIPTION
**Pull request by weshaggard**

### CODE REVIEW

File: eng/pipelines/mirror-repos.yml [hunk 1]
```diff
@@ -109,6 +109,12 @@ jobs:
               Azure/azure-sdk-for-c-pr:
               azure-sdk/azure-sdk-for-c:
 
+          Azure/azure-sdk-for-rust:
+            TargetRepos:
+              Azure/azure-sdk-for-rust-pr:
+              azure-sdk/azure-sdk-for-rust:
+              azure-sdk/azure-sdk-for-rust-pr:
```
Comment (RickWinter) [inline eng/pipelines/mirror-repos.yml:116]: What are these repos for?

**Review by RickWinter — APPROVED** [review]

File: eng/pipelines/mirror-repos.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/mirror-repos.yml:116]: This is our bot fork and we need to those forks for various automation that we have the bot doing. We add them to the mirroring list to keep them updated as well.

Comment (RickWinter) [reply eng/pipelines/mirror-repos.yml:116]: I noticed the repos exist for C++ but it doesn't have the mirroring.  Its 2K changes behind.  Do you want to update as part of this PR as well?

Comment (RickWinter) [reply eng/pipelines/mirror-repos.yml:116]: And C might be missing one as well?
  "azure-sdk/azure-sdk-for-c-pr"
  
  
          Azure/azure-sdk-for-c:
            TargetRepos:
              Azure/azure-sdk-for-c-pr:
              azure-sdk/azure-sdk-for-c:

**Pull request merged**